block.go

```diff
@@ -109,9 +111,10 @@ type StringTuples interface {
 type ChunkWriter interface {
 	// WriteChunks writes several chunks. The Chunk field of the ChunkMetas
 	// must be populated.
+	// b is the byte buffer that can be used by WriteChunks.
 	// After returning successfully, the Ref fields in the ChunkMetas
 	// are set and can be used to retrieve the chunks from the written data.
-	WriteChunks(chunks ...chunks.Meta) error
+	WriteChunks(b []byte, chunks ...chunks.Meta) ([]byte, error)
```

Comment:
Can we comment the returned buffered somehow? I guess it is for better reuse (we reuse full slice not only array).

Also I don't think we need this if we would ensure slice is greater than 5 binary.MaxVarintLen32 here, right? This is the only place we grow the slice.

By look of usage b is already of size b = make([]byte, binary.MaxVarintLen32) in all cases?

Reply:
I will add a comment about it.

And [this comment](https://github.com/prometheus/tsdb/pull/627#discussion_r296443943) for the rest of the part.